### PR TITLE
Better logging of everything

### PR DESCRIPTION
**Pull request by MatthewZMSU**
Better logging of everything

# Description

Currently we did not have proper logging of request and errors. Now we have.

New dependency occured: *Winston* - library for logging

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

We still have no tests for puppeteer-service.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
image_2023-12-01_174043130

### CODE REVIEW

File: helpers/utils.js [hunk 1]
```diff
@@ -171,11 +171,19 @@ exports.performAction = async function performAction(request, action) {
...
+            return {
+                ...response,
+                contextId,
+                pageId,
+            };
         } catch (err) {
-            err.contextId = page.browserContext().id;
-            err.pageId = page.target()._targetId;
+            err.contextId = contextId;
+            err.pageId = pageId;
```
Comment (MatthewZMSU) [inline helpers/utils.js:186]: Here I fix `action` requests from user.
Without it user does not get correct `scrapy-puppeteer-service-context-id` header (see routes/action.js).

Comment (mxsnq) [reply helpers/utils.js:188]: Ok
Do you think we should let user override page id (i.e. reversing to `{contextId, pageId, ...response}`)? It is technically possible to open new page in custom actions. We don't have real usages of it though and we may discourage it.

**Review by mxsnq — APPROVED** [review]
Looks good. I have some additional comments, but I'm ok with merging it as is.

File: helpers/utils.js (hunk 1 above)
Comment (MatthewZMSU) [reply helpers/utils.js:188]: Probably we should ban user from direct return `contextId` and `pageId` from the function as we override them in `performAction`.
Another solution I thought of is to return `{contextId, pageId, response}`. In this case we do not ban user from anything but we need to rewrite some parts in Python.

File: helpers/loggers.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        body = JSON.stringify(body, null, 8);
+        body = body.replace(/\\n/g, "\n");  // To log new lines and not `\n`
+    } else {  // Other requests
+        body = JSON.stringify(body, null, 8);
+    }
+
+    return body;
+}
+
+exports.format = function format(tokens, req, res) {
```
[the remark this answers is not in the document]
Comment (MatthewZMSU) [reply helpers/loggers.js:71]: I have thought of it but I could not implement any non-duplicating solution. May be the problem is in different structure of loggers.
I suggest the problem to be future work.

**Pull request merged**